provisioner/registryv1/README.md

```diff
@@ -0,0 +1,123 @@
+information on all ConfigMaps related to this bundle. 
+
+`kubectl get cm bundle-metadata-my-bundle -o yaml` provides access to this metadata ConfigMap, which then links to all other ConfigMap objects
+where the unpacked data is stored. 
+
+Surfacing the content of a bundle in a more user-friendly way, via a plugin or additional API, is on the RukPak roadmap. 
+
+### Pivot between bundle versions 
+
+The `BundleInstance` API is meant to indicate the version of the bundle that should be active within the cluster. Given two unpacked bundles in the cluster,
+`my-bundle-v0.0.1` and `my-bundle-v0.0.2`, the `spec.bundleName` field of the related `BundleInstance` can be updated from say, the older version,
+to the newer version, and the registryv1 provisioner will be able to successfully create the new content on-cluster and remove the old content.  
+
+## Running locally 
+
+To experiment with the registryv1 provisioner locally, first setup a local cluster. It's recommended to have OLM installed on the cluster as well, 
+which can be done easily by heading to the [releases](https://github.com/operator-framework/operator-lifecycle-manager/releases) section and installing via the install script. 
+
+Once the cluster is setup and OLM is running, take the following steps from the root of the repository:
+* Create the rukpak namespace via `kubectl create ns rukpak-system` by default
+* Run `make bin/registryv1` to create the registryv1 provisioner binary locally
+* Apply the RukPak core CRDs via `kubectl apply -f manifests/`
+* Run `./bin/registryv1` to start the controller 
+
+From there, creating some `Bundles` and `BundleInstance` types to see the provisioner in action. For an example bundle to use, 
+the [combo operator](https://github.com/operator-framework/combo) is a good candidate. 
+
+Create the bundle:
+```yaml
+---
+apiVersion: core.rukpak.io/v1alpha1
+kind: Bundle
+metadata:
+  name: combo-0.0.1
+spec:
+  image: quay.io/operator-framework/combo-bundle:v0.0.1
+  provisionerClassName: core.rukpak.io/registry+v1
+```
+
+Before creating the combo `BundleInstance` on-cluster, ensure that there is a namespace available with an OperatorGroup that 
+supports the combo operator. 
+
+Create the namespace and OperatorGroup:
+```yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: combo-system
+---
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+  name: combo-system-og
+  namespace: combo-system
+spec: {}
+```
+
+Create the combo `BundleInstance` referencing the combo `Bundle` available in the cluster. 
+
+```yaml
+apiVersion: core.rukpak.io/v1alpha1
+kind: BundleInstance
+metadata:
+  name: combo-system
+spec:
+  provisionerClassName: core.rukpak.io/registry+v1
+  bundleName: combo-0.0.1
+```
+
+From there, check out the combo operator and ensure that the CSV is present on the cluster. 
```

Comment:
I like using the combo operator as an example, but when I install on the cluster locally the CSV stays in a `Pending` state. looking at the status it looks like there are some PolicyRules that are missing. This is maybe due to a gap between installing a CSV directly versus via a Subscription to a catalog. Not sure what the next step is. 

```
  - dependents:
    - group: rbac.authorization.k8s.io
      kind: PolicyRule
      message: namespaced rule:{"verbs":["get","list","watch","create","update","patch","delete"],"apiGroups":[""],"resources":["configmaps"]}
      status: NotSatisfied
      version: v1
    - group: rbac.authorization.k8s.io
      kind: PolicyRule
      message: namespaced rule:{"verbs":["get","list","watch","create","update","patch","delete"],"apiGroups":["coordination.k8s.io"],"resources":["leases"]}
      status: NotSatisfied
      version: v1
    - group: rbac.authorization.k8s.io
      kind: PolicyRule
      message: namespaced rule:{"verbs":["create","patch"],"apiGroups":[""],"resources":["events"]}
      status: NotSatisfied
```

Installing a dummy operator instead, something without any permisions or just a set of APIs is possible for the purposes of the README but not as compelling.

Reply:
Combo is pretty tailor made for this purpose so we should try and see what we can do to get around that. Is this an issue with Combo's implementation or with the registryv1 provisioner?

Reply:
Agreed -- I have to look further but I imagine, given that you said you were able to install it successfully via the traditional flow (add the bundle to a catalog, make a subscription, etc), that it's something to do with the provisioner. Not all of the roles/policyrules that are required for combo to work (if the rules I mentioned above are necessary) are being generated.

Reply:
This is definitely the registryv1 provisioner not doing everything that the install plan does. Its the weirdness I was talking about that I hacked around in my demo. I'm pretty sure that the install plan generates and applies these things such that they exist by the time the CSV controller is reconciling the CSV.

Reply:
Yes, seems likely. I guess the question now is whether we want to transfer over that existing functionality to registryv1 provisioner? I imagine the answer is yes if we want to have a useful provisioner. We want to avoid rebuilding OLM of course but if its simply a matter of generating necessary permissions on install it should be doable. 

I'm not sure if there are other gaps around cert creation/handling as well or if webhooks will work alright.

Reply:
See #65 for tracking issue